[2025-07-06T12:31:22.347+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-06T12:31:22.358+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: practice_etl_pipeline.transform_and_load manual__2025-07-06T12:31:11.877799+00:00 [queued]>
[2025-07-06T12:31:22.362+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: practice_etl_pipeline.transform_and_load manual__2025-07-06T12:31:11.877799+00:00 [queued]>
[2025-07-06T12:31:22.362+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-06T12:31:22.370+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): transform_and_load> on 2025-07-06 12:31:11.877799+00:00
[2025-07-06T12:31:22.374+0000] {clientserver.py:570} INFO - Closing down clientserver connection
[2025-07-06T12:31:22.374+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=2675) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-07-06T12:31:22.375+0000] {standard_task_runner.py:63} INFO - Started process 2816 to run task
[2025-07-06T12:31:22.376+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'practice_etl_pipeline', 'transform_and_load', 'manual__2025-07-06T12:31:11.877799+00:00', '--job-id', '268', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmpdtyk93u1']
[2025-07-06T12:31:22.378+0000] {standard_task_runner.py:91} INFO - Job 268: Subtask transform_and_load
[2025-07-06T12:31:22.409+0000] {task_command.py:426} INFO - Running <TaskInstance: practice_etl_pipeline.transform_and_load manual__2025-07-06T12:31:11.877799+00:00 [running]> on host 93b0937d74ff
[2025-07-06T12:31:22.457+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='practice_etl_pipeline' AIRFLOW_CTX_TASK_ID='transform_and_load' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T12:31:11.877799+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-06T12:31:11.877799+00:00'
[2025-07-06T12:31:22.458+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-06T12:31:23.706+0000] {logging_mixin.py:188} INFO - +----------+----------+-----------+-----------+
|   on_date|account_rk|currency_rk|balance_out|
+----------+----------+-----------+-----------+
|31.12.2017|     14138|         44|2.0003622E7|
|31.12.2017|  34156756|         34|   99438.49|
|31.12.2017| 277937699|         34|   35341.17|
|31.12.2017|     13832|         34|    3951.94|
|31.12.2017|     20358|         34|   60328.29|
|31.12.2017|     23329|         21|    6245.06|
|31.12.2017|     17442|         34|   33245.41|
|31.12.2017|     16024|         34|   20673.34|
|31.12.2017|     40266|         34|   35664.89|
|31.12.2017|     13597|         35|   59604.09|
|31.12.2017|     13632|         44|2.0082184E7|
|31.12.2017|     24232|         44|   35263.28|
|31.12.2017|     17441|         34|  1072653.4|
|31.12.2017|     18006|         44|   80422.81|
|31.12.2017|     13843|         34|    17933.5|
|31.12.2017|     17436|         34|5.0050756E7|
|31.12.2017|     17434|         34|  5008134.0|
|31.12.2017|     24235|         35|   81577.79|
|31.12.2017|     24775|         34|   59326.31|
|31.12.2017| 306013958|         34|   63292.91|
+----------+----------+-----------+-----------+
only showing top 20 rows
[2025-07-06T12:31:24.008+0000] {logging_mixin.py:188} INFO - +----------+----------+-----------+-----------+
|on_date   |account_rk|currency_rk|balance_out|
+----------+----------+-----------+-----------+
|2017-12-31|14138     |44         |2.0003622E7|
|2017-12-31|34156756  |34         |99438.49   |
|2017-12-31|277937699 |34         |35341.17   |
|2017-12-31|13832     |34         |3951.94    |
|2017-12-31|20358     |34         |60328.29   |
|2017-12-31|23329     |21         |6245.06    |
|2017-12-31|17442     |34         |33245.41   |
|2017-12-31|16024     |34         |20673.34   |
|2017-12-31|40266     |34         |35664.89   |
|2017-12-31|13597     |35         |59604.09   |
|2017-12-31|13632     |44         |2.0082184E7|
|2017-12-31|24232     |44         |35263.28   |
|2017-12-31|17441     |34         |1072653.4  |
|2017-12-31|18006     |44         |80422.81   |
|2017-12-31|13843     |34         |17933.5    |
|2017-12-31|17436     |34         |5.0050756E7|
|2017-12-31|17434     |34         |5008134.0  |
|2017-12-31|24235     |35         |81577.79   |
|2017-12-31|24775     |34         |59326.31   |
|2017-12-31|306013958 |34         |63292.91   |
+----------+----------+-----------+-----------+
only showing top 20 rows
[2025-07-06T12:31:28.535+0000] {logging_mixin.py:188} INFO - +----------+------+----------------------------+-------+
|timestamp |status|message                     |obj_num|
+----------+------+----------------------------+-------+
|2025-07-06|ИНФО  |создана таблица ft_balance_f|114    |
+----------+------+----------------------------+-------+
[2025-07-06T12:31:28.927+0000] {logging_mixin.py:188} INFO - +----------+-----------------+----------------+-------------+------------+
| oper_date|credit_account_rk|debet_account_rk|credit_amount|debet_amount|
+----------+-----------------+----------------+-------------+------------+
|09-01-2018|        311159924|           13630|      1956.75|    16620.79|
|09-01-2018|            13985|           13630|     28022.45|    46828.89|
|09-01-2018|        393808409|           17436|      14301.0|    37300.36|
|09-01-2018|        311159924|           13630|      6639.01|    14526.86|
|09-01-2018|            17436|           13630|     74112.52|    39235.85|
|09-01-2018|        409685020|           13630|      79622.8|    65112.67|
|09-01-2018|        409685020|           13630|     93991.39|    85537.28|
|09-01-2018|         10006159|           13630|     51468.22|     50535.5|
|10-01-2018|         10006159|           13630|     67206.08|     5379.09|
|10-01-2018|          8736635|           13630|     86567.13|    23882.83|
|10-01-2018|        312741916|           13630|      81153.8|    44204.63|
|10-01-2018|        311159924|           13630|      2722.73|    95152.91|
|10-01-2018|         46179628|           13630|     91585.92|    25820.92|
|10-01-2018|        393808409|           17436|     56189.19|    62507.77|
|10-01-2018|        393808409|           17436|     20321.48|    81677.47|
|11-01-2018|        393808409|           17436|     36589.69|    10583.49|
|11-01-2018|            37051|           13630|     90864.15|    63022.99|
|11-01-2018|            13630|           17436|     69741.71|    15743.36|
|11-01-2018|        393808409|           17436|     14843.08|    58578.34|
|12-01-2018|            37051|           13630|     78799.92|     4133.29|
+----------+-----------------+----------------+-------------+------------+
only showing top 20 rows
[2025-07-06T12:31:29.172+0000] {logging_mixin.py:188} INFO - +----------+-----------------+----------------+-------------+------------+
|oper_date |credit_account_rk|debet_account_rk|credit_amount|debet_amount|
+----------+-----------------+----------------+-------------+------------+
|2018-01-09|311159924        |13630           |1956.75      |16620.79    |
|2018-01-09|13985            |13630           |28022.45     |46828.89    |
|2018-01-09|393808409        |17436           |14301.0      |37300.36    |
|2018-01-09|311159924        |13630           |6639.01      |14526.86    |
|2018-01-09|17436            |13630           |74112.52     |39235.85    |
|2018-01-09|409685020        |13630           |79622.8      |65112.67    |
|2018-01-09|409685020        |13630           |93991.39     |85537.28    |
|2018-01-09|10006159         |13630           |51468.22     |50535.5     |
|2018-01-10|10006159         |13630           |67206.08     |5379.09     |
|2018-01-10|8736635          |13630           |86567.13     |23882.83    |
|2018-01-10|312741916        |13630           |81153.8      |44204.63    |
|2018-01-10|311159924        |13630           |2722.73      |95152.91    |
|2018-01-10|46179628         |13630           |91585.92     |25820.92    |
|2018-01-10|393808409        |17436           |56189.19     |62507.77    |
|2018-01-10|393808409        |17436           |20321.48     |81677.47    |
|2018-01-11|393808409        |17436           |36589.69     |10583.49    |
|2018-01-11|37051            |13630           |90864.15     |63022.99    |
|2018-01-11|13630            |17436           |69741.71     |15743.36    |
|2018-01-11|393808409        |17436           |14843.08     |58578.34    |
|2018-01-12|37051            |13630           |78799.92     |4133.29     |
+----------+-----------------+----------------+-------------+------------+
only showing top 20 rows
[2025-07-06T12:31:41.548+0000] {logging_mixin.py:188} INFO - +----------+------+----------------------------+-------+
|timestamp |status|message                     |obj_num|
+----------+------+----------------------------+-------+
|2025-07-06|ИНФО  |создана таблица ft_posting_f|33892  |
+----------+------+----------------------------+-------+
[2025-07-06T12:31:41.558+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-06T12:31:41.559+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/main.py", line 222, in transform_and_load
    df = spark.read.csv(CSV_PATH + file, header=True, sep=';', schema=BD_SCHEMA[table_name]).distinct()
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/readwriter.py", line 799, in csv
    self._set_opts(
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/readwriter.py", line 50, in _set_opts
    self.schema(schema)  # type: ignore[attr-defined]
    ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/readwriter.py", line 152, in schema
    self._jreader = self._jreader.schema(jschema)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/spark/spark-4.0.0-bin-hadoop3/python/lib/py4j-0.10.9.9-src.zip/py4j/java_gateway.py", line 1362, in __call__
    return_value = get_return_value(
                   ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/errors/exceptions/captured.py", line 288, in deco
    raise converted from None
pyspark.errors.exceptions.captured.AnalysisException: [UNSUPPORTED_CHAR_OR_VARCHAR_AS_STRING] The char/varchar type can't be used in the table schema.
If you want Spark treat them as string type as same as Spark 3.0 and earlier, please set "spark.sql.legacy.charVarcharAsString" to "true". SQLSTATE: 0A000
[2025-07-06T12:31:41.567+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=practice_etl_pipeline, task_id=transform_and_load, run_id=manual__2025-07-06T12:31:11.877799+00:00, execution_date=20250706T123111, start_date=20250706T123122, end_date=20250706T123141
[2025-07-06T12:31:41.585+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 268 for task transform_and_load ([UNSUPPORTED_CHAR_OR_VARCHAR_AS_STRING] The char/varchar type can't be used in the table schema.
If you want Spark treat them as string type as same as Spark 3.0 and earlier, please set "spark.sql.legacy.charVarcharAsString" to "true". SQLSTATE: 0A000; 2816)
[2025-07-06T12:31:41.625+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-06T12:31:41.635+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-06T12:31:41.639+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
